tests/phpunit/CRM/MembershipExtras/Service/MembershipInstalmentsScheduleTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+use CRM_MembershipExtras_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembershipExtras_Service_MembershipInstalmentsSchedule as MembershipInstalmentsSchedule;
+use CRM_MembershipExtras_Exception_InvalidMembershipTypeInstalmentCalculator as InvalidMembershipTypeInstalmentCalculator;
+use CRM_MembershipExtras_Service_MoneyUtilities as MoneyUtilities;
+
+/**
+ * Class CRM_MembershipExtras_Service_MembershipTypeInstalmentsScheduleTest
+ *
+ * @group headless
+ */
+class CRM_MembershipExtras_Service_MembershipTypeInstalmentsScheduleTest extends BaseHeadlessTest {
+
+  /**
+   * Tests Generate Rolling Monthly Instalments
+   *
+   * @throws Exception
+   */
+  public function testGenerateRollingMonthlyInstalments() {
+    $membershipTypes = $this->getRollingMembershipTypes();
+    $instalments = $this->getRollingMembershipInstalments(
+      $membershipTypes,
+      MembershipInstalmentsSchedule::MONTHLY
+    );
+
+    $this->assertCount(MembershipInstalmentsSchedule::MONTHLY_INSTALMENT_COUNT, $instalments);
+
+    $expectedAmount = MoneyUtilities::roundToPrecision(
+      ($membershipTypes[0]->minimum_fee + $membershipTypes[1]->minimum_fee) / MembershipInstalmentsSchedule::MONTHLY_INSTALMENT_COUNT,
+      2
+    );
+
+    $instalment = $instalments[0];
+    $this->assertEquals($expectedAmount, $instalment->getInstalmentAmount()->getAmount());
```

Comment:
you will need to also test that each instalment has the correct date, and I would do it in separate test method, and I will rename this one so you end up with : 

- testRollingMonthlyInstalmentsScheduleAmounts
- testRollingMonthlyInstalmentsScheduleDates

and the same goes for annually and quarterly

Reply:
@omarabuhussein  noted on this, I'll leave annually and quarterly to the next PR as this PR is only related to monthly instalment only.

Reply:
this again has nothing to do with it being annually or quarterly, actually, it even makes more sense to be done for monthly that anything else since you need to check that the 12 instalments dates are correct.